src/core/service.c

```diff
@@ -3894,19 +3894,19 @@ static void service_force_watchdog(Service *s) {
 static void service_notify_message(
                 Unit *u,
                 const struct ucred *ucred,
-                char **tags,
+                char * const *tags,
```

Comment:
we mostly gave up on using `const` on arrays of strings, since the semantics of that are so fucked in C. but i guess it doesn't hurt if it works here

Reply:
I have tried to had a few const at some place, I quickly forget about it. And yes I am agree this part in the C language is "not perfect".